### PR TITLE
Add deployment docs

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: docs/deployment.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Deployment
+
+GLAM has a 3-stage deployment:
+
+1. **Dev**: Deployments to the development server happen automatically for all commits that land on the master branch
```
Comment (hamilton) [inline docs/deployment.md]:
```suggestion
1. **Dev**: Deployments to the development server happen automatically for all commits that land on the `master` branch
```

File: docs/deployment.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+2. **Stage**: Deployments to the staging server happen automatically for tags of the form `m<integer>` or of the [CalVer](http://calver.org/) format `YYYY.MM.N`
```
Comment (hamilton) [inline docs/deployment.md]: is it `YYYY.MM.N`? `N` isn't in the CalVer docs

File: docs/deployment.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+3. **Production**: TBD
+
+## Tagging for release
+
+The steps for tagging are release are as follows:
```
Comment (hamilton) [inline docs/deployment.md]:
```suggestion
The steps for tagging a release for the staging server are as follows:
```

File: docs/deployment.md (hunk 2 above)
Comment (robhudson) [reply docs/deployment.md]: I can document that here. It's what we've done at mozilla on other projects. `N` is a zero-based counter for the number of tags of the month and does not relate to a day.

**Pull request merged**